### PR TITLE
UI improvements

### LINKED ISSUE #242: Improve user feedback
As a user I'd like to know whether actions requiring an API call were successful or not e.g. creating or updating users etc.

### LINKED ISSUE #245: Official finnish localization for DataGrids
As a user I'd like to have the MUI official finnish localization used in all DataGrid components.

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #245 
Resolves #242 

Changes all Data Grid elements localization to MUI official Finnish localization.

Adds a Redux slice for snackbar message and open state to allow displaying feedback of successful actions that result in changing the screen. e.g. restoring an archived draft version, creating/saving/publishing a form.

Also slightly improves error dialog message as discussed with project lead.

### CODE REVIEW

File: src/components/screens/editor/editor-screen.tsx [hunk 1]
```diff
@@ -206,7 +213,7 @@ const EditorScreen: React.FC = () => {
 
     setLoading(false);
   };
-
+
```
Comment (Jdallos) [inline src/components/screens/editor/editor-screen.tsx:216]:
```suggestion
 
```

File: src/components/screens/editor/editor-screen.tsx [hunk 2]
```diff
@@ -239,6 +247,11 @@ const EditorScreen: React.FC = () => {
           </RoundActionButton>
         </NavigationTabContainer>
         <Divider/>
+        <Button
+          onClick={ () => dispatch(setSnackbarMessage("MOI")) }
+        >
+          Click
+        </Button>
```
Comment (Jdallos) [inline src/components/screens/editor/editor-screen.tsx:254]:
```suggestion
        <Divider/>
```

Comment (Jdallos) [reply src/components/screens/editor/editor-screen.tsx:254]: remove button

File: src/components/screens/users/users-screen.tsx [hunk 3]
```diff
@@ -313,7 +322,6 @@ const UsersScreen: React.FC = () => {
    * Event handler for User edit dialog cancel
    */
   const onEditMemberDialogCancel = () => setEditMemberOpen(false);
-
   React.useEffect(() => {
```
Comment (Jdallos) [inline src/components/screens/users/users-screen.tsx:325]:
```suggestion

  React.useEffect(() => {
```

**Review by Jdallos — APPROVED** [review]
With comments

File: src/components/screens/users/users-screen.tsx (hunk 3 above)
Comment (EVP1989) [reply src/components/screens/users/users-screen.tsx:325]: Also move useEffect to React import

File: src/localization/fi.json [hunk 4]
```diff
@@ -385,7 +387,12 @@
       "lastNameLabel": "Sukunimi",
       "upnNumberLabel": "UPN-numero",
       "cancelButton": "Peruuta",
-      "editButton": "Muokkaa käyttäjää"
+      "editButton": "Muokkaa käyttäjää",
+      "snackbarText": "Käyttäjän muokkaaminen onnistui!"
+    },
+    "groupMembership": {
+      "groupMembershipAddSnackbarText": "Käyttäjä lisääminen ryhmään onnistui!",
```
Comment (EVP1989) [inline src/localization/fi.json:394]:
```suggestion
      "groupMembershipAddSnackbarText": "Käyttäjän lisääminen ryhmään onnistui!",
```

File: src/localization/fi.json [hunk 5]
```diff
@@ -531,6 +538,30 @@
 
   "dataGrid": {
     "rowsPerPage": "Rivejä sivulla:"
-  }
+  },
 
+  "successSnackbars": {
+    "formEditor": {
+      "deleteFormSuccessText": "Tuotantolomake poistettu onnistuneesti!",
+      "deleteFormVersionSuccessText": "Lomake luonnos poistettu onnistuneesti!",
```
Comment (EVP1989) [inline src/localization/fi.json:546]:
```suggestion
      "deleteFormVersionSuccessText": "Lomakkeen luonnos poistettu onnistuneesti!",
```

File: src/localization/fi.json [hunk 6]
```diff
@@ -531,6 +538,30 @@
...
+      "createFormSuccessText": "Uusi lomake luotu onnistuneesti!",
+      "restoreArchiveFormSuccessText": "Arkistoitu lomake palautettu onnistuneesti!",
+      "convertSchemaSuccessText": "Sosmeta skeema konvertoitu onnistuneesti!"
```
Comment (EVP1989) [inline src/localization/fi.json:549]:
```suggestion
      "convertSchemaSuccessText": "Sosmetan lomakepohja luotu onnistuneesti!"
```

File: src/localization/fi.json [hunk 7]
```diff
@@ -531,6 +538,30 @@
...
+    },
+    "draftEditor": {
+      "saveDraftSuccessText": "Lomake luonnos tallennettu onnistuneesti!",
```
Comment (EVP1989) [inline src/localization/fi.json:552]:
```suggestion
      "saveDraftSuccessText": "Lomakkeen luonnos tallennettu onnistuneesti!",
```

File: src/localization/fi.json [hunk 8]
```diff
@@ -531,6 +538,30 @@
...
+    "users": {
+      "addMemberSuccessText": "Käyttäjän lisääminen onnistui!",
+      "editUserSuccessText": "Käyttäjän muokkaaminen onnistui!",
+      "addMemberGroupSuccessText": "Käyttäjäryhmän lisääminen onnistui!",
+      "groupMembershipAddSuccessText": "Käyttäjän lisääminen ryhmään onnistui!",
+      "groupMembershipRemoveSuccessText": "Käyttäjän poistaminen ryhmästä onnistui!"
+    },
+    "replies": {
+      "replyDeleteSuccessText": "Lomakkeen poistaminen onnistui!",
+      "replyEditSuccessText": "Lomakkeen muokkaus onnistui!"
```
Comment (EVP1989) [inline src/localization/fi.json:564]:
```suggestion
      "replyDeleteSuccessText": "Vastauksen poistaminen onnistui!",
      "replyEditSuccessText": "Vastauksen muokkaus onnistui!"
```

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/screens/editor/editor-screen.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/screens/editor/editor-screen.tsx:216]: Resolved

File: src/components/screens/editor/editor-screen.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/screens/editor/editor-screen.tsx:254]: Resolved

File: src/components/screens/users/users-screen.tsx (hunk 3 above)
Comment (villejuutila) [reply src/components/screens/users/users-screen.tsx:325]: Resolved

File: src/localization/fi.json (hunk 4 above)
Comment (villejuutila) [reply src/localization/fi.json:399]: Resolved

File: src/localization/fi.json (hunk 5 above)
Comment (villejuutila) [reply src/localization/fi.json:551]: Resolved

File: src/localization/fi.json (hunk 6 above)
Comment (villejuutila) [reply src/localization/fi.json:554]: Resolved

File: src/localization/fi.json (hunk 7 above)
Comment (villejuutila) [reply src/localization/fi.json:557]: Resolved

File: src/localization/fi.json (hunk 8 above)
Comment (villejuutila) [reply src/localization/fi.json:569]: Resolved

**Pull request merged**